src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ExperimentalReadThreadingGraph.java

```diff
@@ -0,0 +1,796 @@
+
+        // create all of the equivalent seq graph vertices
+        for ( final MultiDeBruijnVertex dv : vertexSet() ) {
+            final SeqVertex sv = new SeqVertex(dv.getAdditionalSequence(isSource(dv)));
+            sv.setAdditionalInfo(dv.getAdditionalInfo());
+            vertexToOrigionalSeqVertex.put(dv, sv);
+            printingSeqGraph.addVertex(sv);
+        }
+
+        // walk through the nodes and connect them to their equivalent seq vertices
+        for( final MultiSampleEdge e : edgeSet() ) {
+            final SeqVertex seqInV = vertexToOrigionalSeqVertex.get(getEdgeSource(e));
+            final SeqVertex seqOutV = vertexToOrigionalSeqVertex.get(getEdgeTarget(e));
+            printingSeqGraph.addEdge(seqInV, seqOutV, new BaseEdge(e.isRef(), e.getMultiplicity()));
+        }
+
+        final Map<SeqVertex, SeqVertex> originalSeqVertexToMergedVerex = printingSeqGraph.zipLinearChainsWithVertexMapping();
+        printingSeqGraph.originalSeqVertexToMergedVerex = originalSeqVertexToMergedVerex;
+        printingSeqGraph.vertexToOrigionalSeqVertex = vertexToOrigionalSeqVertex;
+
+
+        printingSeqGraph.printGraph(graphWriter, writeHeader, pruneFactor);
+    }
+
+    /**
+     * Walk along the reference path in the graph and pull out the corresponding bases
+     *
+     * NOTE: this attempts to generate the longest sequence of refernce bases in the event that fromVertex or toVertex are non-unique
+     *
+     * @param fromVertex    starting vertex
+     * @param toVertex      ending vertex
+     * @param includeStart  should the starting vertex be included in the path
+     * @param includeStop   should the ending vertex be included in the path
+     * @return              byte[] array holding the reference bases, this can be null if there are no nodes between the starting and ending vertex (insertions for example)
+     */
+    @Override
+    public byte[] getReferenceBytes( final MultiDeBruijnVertex fromVertex, final MultiDeBruijnVertex toVertex, final boolean includeStart, final boolean includeStop ) {
+        Utils.nonNull(fromVertex, "Starting vertex in requested path cannot be null.");
+        Utils.nonNull(toVertex, "From vertex in requested path cannot be null.");
+
+        byte[] bytes = null;
+        int fromIndex = referencePath.indexOf(fromVertex);
+        int toIndex = referencePath.lastIndexOf(toVertex);
+
+        if( includeStart ) {
+            bytes = ArrayUtils.addAll(bytes, getAdditionalSequence(fromVertex, true));
+        }
+        for (int i = fromIndex + 1; i < toIndex; i++) {
+            bytes = ArrayUtils.addAll(bytes, getAdditionalSequence(referencePath.get(i)));
+        }
+
+        if( includeStop ) {
+            bytes = ArrayUtils.addAll(bytes, getAdditionalSequence(toVertex));
+        }
+        return bytes;
+    }
+
+    // Generate threading trees
+    public void generateJunctionTrees() {
+        buildGraphIfNecessary();
```

Comment:
I'm actually okay with leaving this. Generating junction trees is a terminal action that requires the graph be "completed". The method itself does nothing in the event the graph has already been constructed so its safe to call this. (it was mostly added for tests anyway so i could be persuaded otherwise)

Reply:
I think I would prefer to throw an `IllegalStateException` if the graph hasn't been built.  The code could get hard to follow if the graph ends up being built as a side effect here.

Reply:
Sure. Ive just updated everything in the tests.